diff --git a/node_modules/oauth4webapi/build/index.js b/node_modules/oauth4webapi/build/index.js
index 4a178ba..746aa6f 100644
--- a/node_modules/oauth4webapi/build/index.js
+++ b/node_modules/oauth4webapi/build/index.js
@@ -1030,10 +1030,15 @@ function validateOptionalIssuer(expected, result) {
     return result;
 }
 function validateIssuer(expected, result) {
-    if (result.claims.iss !== expected) {
-        throw new OPE('unexpected JWT "iss" (issuer) claim value');
-    }
-    return result;
+  if (expected === 'https://login.microsoftonline.com/common/v2.0'
+    && result.claims.tid !== undefined
+    && result.claims.iss === `https://login.microsoftonline.com/${result.claims.tid}/v2.0`) {
+      return result;
+  }
+  if (result.claims.iss === expected) {
+    return result
+  }
+  throw new OPE('unexpected JWT "iss" (issuer) claim value')
 }
 const branded = new WeakSet();
 function brand(searchParams) {
